recipes/alevin-fry/use-libradicl-0.8.2.patch

```diff
@@ -0,0 +1,13 @@
+diff --git i/Cargo.toml w/Cargo.toml
+index eeefa95..e8a7b28 100644
+--- i/Cargo.toml
++++ w/Cargo.toml
+@@ -35,7 +35,7 @@ categories = ["command-line-utilities", "science"]
+ [dependencies]
+ # for local development, look in the libradicl git repository
+ # but when published, pull the specified version
+-libradicl = { git = "https://github.com/COMBINE-lab/libradicl", branch = "develop", version = "0.8.2" }
++libradicl = "0.8.2"
```

Comment:
@rob-p It seems libradicl is at crates.io for some time. Would it be OK to change this upstream ?

Reply:
Hi @martin-g : This is present upstream in crates.io, yes.   However, this line in the Cargo.toml is important in the repo because of the behavior of Cargo.

Specifically, when building locally (i.e. during development) cargo will pull from the repo.  However, when pushing to crates.io, Cargo cannot depend on a crate in a GitHub repo, so it will look for (and pull down) the official crate as long as it's version compatible.

This is, in fact, exactly the behavior we want as libradicl is still under development and improvement (and alevin-fry is as well).  This allows us to tag and have a newer version of libradicl on git during development and then publish to crates.io when we are ready to bump alevin-fry.

Actually, if the stated version is available on crates.io, I'm not sure what the behavior of this is.  Perhaps there is a cargo flag to prefer the crates.io version, rather than the github version?

Reply:
I agree that this works well for development (`develop` branch).
But when making a release (a Git tag) you could set the version and use crates.io instead of Git branch.

Reply:
That's a good suggestion, but I'll have to figure out how to automate this.  Having to manually change the source of crates (right now it's only one but I imagine there could be other crates in active development by the lab on which different tools rely) back and forth between release / develop could become kind of error prone.  Ideally this type of "textual replacement" in the `Cargo.toml` is something that could be done by an appropriate script and run on the GitHub CI every time a new release is cut.